src/ctia/stores/es/sort.clj

```diff
@@ -18,6 +27,35 @@
       ;; https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html#_sort_values
       {field-name {:order order}}
 
+      ;; chooses the maximum value in field (a list) based on remappings,
+      ;; and sorts based on that value for each entity. Remappings must
+      ;; be positive integers.
+      ;; eg
+      ;; {:op :remap-list-max
+      ;;  :field-name "tactics"
+      ;;  :remappings {"TA0043" 1
+      ;;               "TA0042" 2}
+      ;;  :sort_order :asc}
+      :remap-list-max
+      (let [{:keys [remap-default remappings]} params
+            remappings (normalize-remappings remappings)]
+        (assert (every? (complement neg?) (vals remappings)) ":remap-list-max :remappings values must be non-negative")
+        {:_script
+         {:type "number"
+          :script {:lang "painless"
+                   :inline (str/join
+                             "\n"
+                             ["if (!doc.containsKey(params.fieldName)) { return params.default }"
+                              "double score = params.default;"
+                              "for (int i = 0; i < doc[params.fieldName].length; ++i) {"
```

Comment:
I could totally imagine a future dev to introduce a typo here, for example writing `params.fiedlName` instead of `params.fieldName`.
Would that be catch before deployment or `params.fiedlName` would be silently `null`?

If that's the case, perhaps it would be nice to have something that could make a quick and dirty parsing and check that every `params.X` has an associated field in the `:params`.

Reply:
I believe painless catches non-existent fields. It's similar to Java in that way.

https://www.elastic.co/guide/en/elasticsearch/painless/master/painless-operators-reference.html#field-access-operator

Reply:
Hopefully the future dev will write unit tests like I have here, that's the only way I have confidence in these scripts.